### DISCUSSION TITLE
Survey-ranking plugin not working

### DISCUSSION THREAD

**Discussion by kaleidosc0pic** [jspsych/jsPsych#3279] [Q&A]
Hi, I'm trying to use the survey-ranking plugin in an experiment. All the other survey plugins seem to work just fine (like the 'trial' var I included below), but when I try to add this 'difficulty' var to the timeline it doesn't show up in the experiment. Am I doing something wrong or is it malfunctioning? I noticed that it didn't have a demo page on the jsPsych website like all the other plugins.

I'm also using version 7.1 if that changes anything.

```
var difficulty = {
  type: jsPsychSurveyRanking,
  prompt: "Please order these by difficulty, with the easiest at the top.",
  options: [
    "1",
    "2",
    "3",
    "4",
    "5",
    "6"
  ],
};

var trial = {
  type: jsPsychSurveyText,
  questions: [
    {prompt: 'What is your date of birth?', placeholder: 'mm/dd/yyyy', required: true},
    {prompt: 'What country do you currently live in?'}
  ],
};
```

**Comment by Shaobin-Jiang** [comment]
That is because ranking is not a standalone plugin, but a supported question type under the [survey](https://www.jspsych.org/7.3/plugins/survey/) plugin.

**Reply by kaleidosc0pic** [reply]
How would I use it in the code?

**Reply by Shaobin-Jiang** [reply]
Read through the docs for yourself. The documentation is quite clear on that point.

**Comment by becky-gilbert** [comment]
Hi @geneva-x, here's how to set up your ranking question using the latest version of the [`survey` plugin](https://www.jspsych.org/7.3/plugins/survey/):

```javascript
var difficulty = {
  type: jsPsychSurvey,
  survey_json: {
    elements: [{
      type: "ranking",
      name: "ranking-question",
      title: "Please order these by difficulty, with the easiest at the top.",
      choices: ["1","2","3","4","5","6"]
    }],
    showQuestionNumbers: false
  }
};
```
Does that work for you?

You can find more examples in the [Examples section](https://www.jspsych.org/7.3/plugins/survey/#examples) on the `survey` plugin docs page, in the `survey` plugin [examples folder](https://github.com/jspsych/jsPsych/tree/main/packages/plugin-survey/examples), and on this new documentation page called "[Building Surveys in jsPsych](https://www.jspsych.org/7.3/overview/building-surveys/)".

You can also find documentation and examples for the ranking question type in the SurveyJS documentation here: https://surveyjs.io/form-library/examples/add-ranking-question-to-form/jquery.

Let us know if you run into problems or have more questions!
